contracts/EtomicSwapV2.sol

```diff
@@ -0,0 +1,481 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract EtomicSwapV2 {
+    enum MakerPaymentState {
+        Uninitialized,
+        PaymentSent,
+        TakerSpent,
+        MakerRefunded
+    }
+
+    struct MakerPayment {
+        bytes20 paymentHash;
+        uint32 paymentLockTime;
+        MakerPaymentState state;
+    }
+
+    event MakerPaymentSent(bytes32 id);
+    event MakerPaymentSpent(bytes32 id, bytes32 secret);
+    event MakerPaymentRefundedTimelock(bytes32 id);
+    event MakerPaymentRefundedSecret(bytes32 id, bytes32 secret);
+
+    mapping(bytes32 => MakerPayment) public maker_payments;
```

Comment:
`maker_payments` should be mixedCase

Reply:
I got used to naming vars in a "Rust way", will fix shortly :slightly_smiling_face: